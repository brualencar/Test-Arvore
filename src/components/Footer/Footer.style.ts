import { styled } from 'styled-components';

export const Wrapper = styled.div``;

export const Container = styled.div`
  display: grid;
  align-items: center;
  left: 0;
  bottom: 0;
  width: 100%;
  font-family: ${({ theme }) => theme.font.family};
  height: 100px;
  border: 0.5px solid ${({ theme }) => theme.colors.border.footer};
  background: ${({ theme }) => theme.colors.background.neutral.white};
  padding: ${({ theme }) => theme.spacing.km10};
  margin-top: ${({ theme }) => theme.spacing.km10};
`;

export const Text = styled.div`
  color: ${({ theme }) => theme.colors.text.neutral.light};
  font-size: ${({ theme }) => theme.font.size.m2};
  font-style: normal;
  font-weight: 500;
`;

export const Button = styled.div`
  gap: ${({ theme }) => theme.spacing.km4};
  justify-content: center;
  display: flex;

  @media only screen and (max-width: 992px) {
    display: none;
  }
`;
