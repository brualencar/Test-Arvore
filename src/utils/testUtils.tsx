import React, { PropsWithChildren } from 'react';

import { ThemeProvider } from 'styled-components';

import { render as rtlRender, RenderOptions } from '@testing-library/react';

import { theme } from '../assets/styles/globalStyle';

function render(ui: React.ReactElement, renderOptions?: RenderOptions) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
