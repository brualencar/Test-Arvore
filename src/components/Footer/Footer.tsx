import React from 'react';
import { Col, Grid, Row } from 'react-flexbox-grid';

import { Button } from '../Button';
import * as Styled from './Footer.style';

function Footer() {
  return (
    <Styled.Wrapper data-testid='footer'>
      <Styled.Container>
        <Grid fluid>
          <Row
            middle='xs'
            center='xs'
            between='xs'>
            <Col
              xl={6}
              lg={6}
              md={12}
              sm={12}
              xs={12}>
              <Styled.Text>
                Copyright © 2023 Árvore. Todos os direitos reservados.
              </Styled.Text>
            </Col>

            <Col
              xl={6}
              lg={6}
              md={2}
              sm={3}
              xs={4}>
              <Styled.Button>
                <Button
                  text='Política de privacidade'
                  variant='primary'
                />
                <Button
                  text='Ajuda'
                  variant='primary'
                />
              </Styled.Button>
            </Col>
          </Row>
        </Grid>
      </Styled.Container>
    </Styled.Wrapper>
  );
}

export default Footer;
