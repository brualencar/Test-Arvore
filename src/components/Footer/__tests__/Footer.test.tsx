import React from 'react';

import { render, screen } from '../../../utils/testUtils';
import Footer from '../Footer';

jest.mock('react-flexbox-grid', () => ({
  Grid: ({ ...props }) => {
    const tempProps = { ...props, fluid: 'demo' };
    return (
      <div
        data-testid='grid'
        {...tempProps}
      />
    );
  },
  Row: ({ ...props }) => {
    const tempProps = { ...props };

    return (
      <div
        data-testid='row'
        {...tempProps}
      />
    );
  },
  Col: ({ ...props }) => {
    const tempProps = { ...props };

    return (
      <div
        data-testid='col'
        {...tempProps}
      />
    );
  },
}));

describe('Footer', () => {
  it('should render snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it('Should render Footer component', () => {
    render(<Footer />);
    const divElement = screen.getByTestId('footer');
    expect(divElement).toBeInTheDocument();
  });
});
