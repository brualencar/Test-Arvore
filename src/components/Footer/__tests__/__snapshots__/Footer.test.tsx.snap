// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render snapshot 1`] = `
<div>
  <div
    class="sc-gtcAbF"
    data-testid="footer"
  >
    <div
      class="sc-dkXrac giTvmx"
    >
      <div
        data-testid="grid"
        fluid="demo"
      >
        <div
          between="xs"
          center="xs"
          data-testid="row"
          middle="xs"
        >
          <div
            data-testid="col"
            lg="6"
            md="12"
            sm="12"
            xl="6"
            xs="12"
          >
            <div
              class="sc-hKVryf dtOJXn"
            >
              Copyright © 2023 Árvore. Todos os direitos reservados.
            </div>
          </div>
          <div
            data-testid="col"
            lg="6"
            md="2"
            sm="3"
            xl="6"
            xs="4"
          >
            <div
              class="sc-eCQiwC biyLda"
            >
              <button
                class="sc-bdDrbm dGflXy"
                type="button"
              >
                Política de privacidade
              </button>
              <button
                class="sc-bdDrbm dGflXy"
                type="button"
              >
                Ajuda
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
